client/state/plugins/installed/selectors.js

```diff
@@ -60,6 +67,12 @@ export function getPlugins( state, siteIds, pluginFilter ) {
 				return memo;
 			}
 
+			// We currently support fetching plugins per site and also fetching all plugins
+			// in bulk, aiming to optimize the UX in some flows.
+			if ( isRequestingForAllSites( state ) ) {
```

Comment:
It looks like `getPluginOnSite` (for example) reuses `getPlugins` behind the scenes, so I think we should be good here. 

I also tested various flows, and I couldn't find any regression in terms of missing loading indicators (or something similar) because of that change.